prov/efa/src/efa.h

```diff
@@ -320,6 +324,16 @@ int efa_cq_open(struct fid_domain *domain_fid, struct fi_cq_attr *attr,
 int efa_av_insert_addr(struct efa_av *av, struct efa_ep_addr *addr,
 		       fi_addr_t *fi_addr, uint64_t flags, void *context);
 
+/* MR functions */
+int efa_mr_reg_rxr_domain(struct fid *fid, const void *buf, size_t len,
```

Comment:
The only difference between efa_mr_reg and efa_mr_reg_rxr_domain is the input fid. In the case of efa_mr_reg_rxr_domain the fid *fid is for rxr_domain and for efa_mr_reg the fid *fid is for efa_domain

Reply:
Right. But if the app calls fi_mr_regv(regattr), which points to efa_mr_regv(regattr) (https://github.com/ofiwg/libfabric/pull/5664/files#diff-d2e75ef5e2f666e01e77c3158b42137aL252), we still have to convert from rxr_domain's fid to efa_domain's fid. So I think we can do the conversion within function efa_mr_reg*(rxr_domain's fid)

Reply:
Right, so I already have a check for this here
https://github.com/ofiwg/libfabric/pull/5664/files#diff-f4d51ce61f103c6000ca0806199b8873R412
This will ensure the call is directed to the right function